dace/memlet.py

```diff
@@ -22,90 +24,196 @@ class Memlet(object):
-                 data,
-                 num_accesses,
-                 subset,
-                 vector_length,
-                 wcr=None,
-                 other_subset=None,
-                 debuginfo=None,
-                 wcr_conflict=True):
-        """ Constructs a Memlet.
-            :param data: The data object or name to access. B{Note:} this
-                         parameter will soon be deprecated.
-            @type data: Either a string of the data descriptor name or an
-                        AccessNode.
-            :param num_accesses: The number of times that the moved data
-                                 will be subsequently accessed. If
-                                 `dace.dtypes.DYNAMIC` (-1),
-                                 designates that the number of accesses is
-                                 unknown at compile time.
-            :param subset: The subset of `data` that is going to be accessed.
-            :param vector_length: The length of a single unit of access to
-                                  the data (used for vectorization
-                                  optimizations).
-            :param wcr: A lambda function specifying how write-conflicts
-                        are resolved. The syntax of the lambda function receives two elements: `current` value and `new` value,
-                        and returns the value after resolution. For example,
-                        summation is `lambda cur, new: cur + new`.
-            :param other_subset: The reindexing of `subset` on the other
-                                 connected data.
-            :param debuginfo: Source-code information (e.g., line, file)
-                              used for debugging.
-            :param wcr_conflict: If False, forces non-locked conflict
-                                 resolution when generating code. The default
-                                 is to let the code generator infer this
-                                 information from the SDFG.
+                 expr: str = None,
+                 data: str = None,
+                 subset: Union[str, subsets.Subset] = None,
+                 other_subset: Union[str, subsets.Subset] = None,
+                 veclen: int = 1,
+                 volume: Union[int, str, symbolic.SymbolicType] = None,
+                 dynamic: bool = False,
+                 wcr: Union[str, ast.AST] = None,
+                 debuginfo: dtypes.DebugInfo = None,
+                 wcr_nonatomic: bool = False,
+                 allow_oob: bool = False):
+        """ 
+        Constructs a Memlet.
+        :param expr: A string expression of the this memlet, given as an ease
+                     of use API. Must follow one of the following forms:
+                     1. ``ARRAY``,
+                     2. ``ARRAY[SUBSET]``,
+                     3. ``ARRAY[SUBSET] -> OTHER_SUBSET``.
+        :param data: (DEPRECATED) Data descriptor name attached to this memlet.
```

Comment:
:pagesofohdeer: Want to add a deprecated warning?

Reply:
Only that parameter is deprecated. I don't know how to add a deprecated parameter.